tests/unit/s2n_random_test.c

```diff
@@ -142,6 +146,110 @@ static S2N_RESULT s2n_basic_pattern_tests(S2N_RESULT (*s2n_get_random_data_cb)(s
     return S2N_RESULT_OK;
 }
 
+static void swap(size_t i, size_t j,
+    uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS]) {
+    uint64_t temp = range_results[j];
+    range_results[j] = range_results[i];
+    range_results[i] = temp;
+}
+
+static void sort_array(uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS]) {
+    for (size_t i = 1; i < NUMBER_OF_RANGE_FUNCTION_CALLS; i++) {
+        for (size_t j = i; j > 0; j--) {
+            if (range_results[j - 1] > range_results[j]) {
+                swap(j - 1, j, range_results);
+            }
+            else {
+                /* Invariant: subarray [0, i-1] is sorted */
+                continue;
+            }
+        }
+    }
+}
+
+static S2N_RESULT s2n_tests_get_range(S2N_RESULT (*s2n_get_range_cb)(int64_t bound, uint64_t *output))
+{
+    uint64_t range_results[NUMBER_OF_RANGE_FUNCTION_CALLS] = {0};
+    uint64_t current_bound;
+    uint64_t current_output;
+    /* The type of the `bound` parameter in s2n_public_random() is signed */
+    int64_t chosen_upper_bound;
+    struct s2n_blob upper_bound_blob = { .data = (void *) &chosen_upper_bound, .size = sizeof(chosen_upper_bound) };
+
+    /* 0 is not a legal upper bound */
+    current_bound = 0;
+    EXPECT_ERROR(s2n_get_range_cb(current_bound, &current_output));
```

Comment:
It's a way of specifying what we expect the error to be. Otherwise a function could be erroring for some other random reason and we wouldn't know. Additionally it provides context for the reviewer about why we expect this test to fail.